### PR TITLE
Add EventGrid samples and readme

### PR DESCRIPTION
**Pull request by pakrym**
cc @kasobol-msft @fabiocav

### CODE REVIEW

File: sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/README.md [hunk 1]
```diff
@@ -1,34 +1,113 @@
...
+```bash
+az eventgrid topic key list --name <your-resource-name> --resource-group <your-resource-group-name> --query "key1"
+```
 
 ## Key concepts
 
-TODO
+### Using Event Grid binding
+
+Please follow the [binding tutorial](https://docs.microsoft.com/azure/azure-functions/functions-bindings-event-grid-trigger?tabs) to lean about using this extension for publishing EventGrid events.
```
Comment (kasobol-msft) [inline sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/README.md:38]: typo 'lean'

**Review by kasobol-msft — APPROVED** [review]

File: sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/README.md (hunk 1 above)
Comment (pakrym) [inline sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/README.md:38]:
```suggestion
Please follow the [binding tutorial](https://docs.microsoft.com/azure/azure-functions/functions-bindings-event-grid-trigger?tabs) to learn about using this extension for publishing EventGrid events.
```

File: sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/README.md [hunk 2]
```diff
@@ -1,34 +1,113 @@
...
+
+The access key can also be found through the [portal](https://docs.microsoft.com/azure/event-grid/get-access-keys), or by using the Azure CLI snippet below:
+```bash
+az eventgrid topic key list --name <your-resource-name> --resource-group <your-resource-group-name> --query "key1"
+```
 
 ## Key concepts
 
-TODO
+### Using Event Grid binding
```
Comment (ellismg) [inline sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/README.md:36]: should we say "output binding"?

**Review by ellismg — APPROVED** [review]
LGTM

File: sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/README.md (hunk 2 above)
Comment (pakrym) [inline sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/README.md:36]:
```suggestion
### Using Event Grid output binding
```

File: sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs [hunk 3]
```diff
@@ -42,7 +43,7 @@ internal EventGridExtensionConfigProvider(Func<EventGridAttribute, IAsyncCollect
         // default constructor
         public EventGridExtensionConfigProvider(ILoggerFactory loggerFactory)
         {
-            _converter = (attr => new EventGridAsyncCollector(new EventGridPublisherClient(new Uri(attr.TopicEndpointUri), new EventGridSharedAccessSignatureCredential(attr.TopicKeySetting))));
+            _converter = (attr => new EventGridAsyncCollector(new EventGridPublisherClient(new Uri(attr.TopicEndpointUri), new AzureKeyCredential(attr.TopicKeySetting))));
```
Comment (ellismg) [inline sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs:46]: Unrelated to this PR: Is there something we could have done in the EventGrid SDK Documentation itself to make it clear which of these you needed to use?

**Pull request merged**